### ISSUE TITLE
Is it time to retire NPN?

### ISSUE THREAD

**Issue by davidben** [netty/netty-tcnative#856]
The ALPN RFC is coming up on ten years old now. The oldest OpenSSL release with ALPN support, 1.0.2, came out January 2015, nine years ago. The oldest OpenSSL without ALPN, 1.0.1, went EOL in December 2016, now seven years ago.

Perhaps it's time to retire NPN support in Netty?

**Comment by davidben** [comment]
I was just taking stock of who was still using NPN APIs to see how close we are to being able to remove them in BoringSSL. Quite a few callers remaining but I figured I'd file an issue and see if you all still need it.

**Comment by normanmaurer** [comment]
I mean we support it in netty and I think we can't remove it without a major version release. Not sure its used in the wild tho

**Comment by davidben** [comment]
I would hope that, by now, it's not load-bearing. But yeah, makes sense. Maybe time to mark deprecated if we think it's unused then?

**Comment by normanmaurer** [comment]
@davidben yeah I think that is definitely something we should do... Will keep you posted
